src/Controllers/ElementalAreaController.php

```diff
@@ -29,23 +33,217 @@ class ElementalAreaController extends CMSMain
 
     private static $url_handlers = [
         'elementForm/$ItemID' => 'elementForm',
-        'POST api/saveForm/$ID' => 'apiSaveForm',
         '$FormName/field/$FieldName' => 'formAction',
+        'DELETE api/archive/$ID!' => 'apiArchive',
+        'GET api/readElementalArea/$elementalAreaID!' => 'apiReadElementalArea',
+        'POST api/create' => 'apiCreate',
+        'POST api/duplicate' => 'apiDuplicate',
+        'POST api/publish' => 'apiPublish',
+        'POST api/saveForm/$ID' => 'apiSaveForm',
+        'POST api/sort' => 'apiSort',
+        'POST api/unpublish' => 'apiUnpublish',
     ];
 
     private static $allowed_actions = [
         'elementForm',
-        'apiSaveForm',
         'formAction',
+        'apiCreate',
+        'apiArchive',
+        'apiDuplicate',
+        'apiPublish',
+        'apiReadElementalArea',
+        'apiSaveForm',
+        'apiSort',
+        'apiUnpublish',
     ];
 
+    public function apiArchive(HTTPRequest $request): HTTPResponse
+    {
+        $this->checkJsonCsrfToken($request);
+        $id = (int) $request->param('ID');
+        $element = BaseElement::get()->byID($id);
+        if (!$element) {
+            $this->jsonError(400);
+        }
+        // There is no check if the Versioned extension is not applied and then checking
+        // canDelete() instead, because elemental does not officially support unversioned elements
+        if (!$element->canDelete()) {
+            $this->jsonError(403);
+        }
+        $element->doArchive();
```

Comment:
```suggestion
        if (!$element->canDelete()) {
            $this->jsonError(403);
        }
        // There is no check if the Versioned extension is not applied and then calling
        // delete() instead, because elemental does not officially support unversioned elements
        $element->doArchive();
```
Comment was in the wrong place about the wrong method call.

Reply:
Updated with a slightly different comment